tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -2212,3 +2214,67 @@ def benford_correlation(x):
     # np.corrcoef outputs the normalized covariance (correlation) between benford_distribution and data_distribution.
     # In this case returns a 2x2 matrix, the  [0, 1] and [1, 1] are the values between the two arrays
     return np.corrcoef(benford_distribution, data_distribution)[0, 1]
+
+
+@set_property("fctype", "combiner")
+def matrix_profile(x, param):
+    """
+    TODO: Documentation
+
+    :param x: the time series to calculate the feature of
+    :type x: numpy.ndarray
+    :param param: contains dictionaries {"sample_pct": x, "threshold": y, "feature": z}
+                  with sample_pct and threshold being parameters of the matrixprofile
+                  package https://matrixprofile.docs.matrixprofile.org/api.html#matrixprofile-compute
+                  and feature being one of "min", "max", "mean", "median", "25", "75"
+                  and decides which feature of the matrix profile to extract
+    :type param: list
+    :return: the different feature values
+    :return type: pandas.Series
+    """
+    x = np.asarray(x)
+
+    def _calculate_pmp(**kwargs):
```

Comment:
Based on the conversations we've been having, we'll use the matrix profile with the maximum subsequence if no window is specified in kwargs. Otherwise, we'll directly calculate the mp with the user specified window value.

```
if 'windows' in kwargs.keys():
    m_p = mp.compute(x,**kwargs)

else:
    m_p = mp.algorithms.maximum_subsequence(x, include_pmp=True)['pmp'][-1]
```
`

Reply:
`'windows' in kwargs `means the same thing and is more pythonic